client/components/domains/register-domain-step/index.jsx

```diff
@@ -150,14 +114,14 @@ const RegisterDomainStep = React.createClass( {
 			notice: null,
 			searchResults: null,
 		};
-	},
+	}
 
 	getNewRailcarSeed() {
 		// Generate a 7 character random hash on base16. E.g. ac618a3
 		return Math.floor( ( 1 + Math.random() ) * 0x10000000 )
 			.toString( 16 )
 			.substring( 1 );
-	},
+	}
 
 	componentWillReceiveProps( nextProps ) {
```

Comment:
`getInitialState`, which you've removed, is used in this function.
Try switching a site while on this view to trigger the error.